lib/Dialect/RTL/Ops.cpp

```diff
@@ -159,6 +160,7 @@ OpFoldResult ExtractOp::fold(ArrayRef<Attribute> operands) {
 
 OpFoldResult AndOp::fold(ArrayRef<Attribute> operands) {
   auto size = inputs().size();
+  assert(size > 0 && "rtl.and should take 1 or more operands");
```

Comment:
Nit, but this should go into the verification hook for `rtl.and`.  As a separate patch, could you write a testcase (in .mlir file) that is invalid and make sure the verifier catches this?  If not, we should add verification support.  @amaleewilson may be more familiar with this in MLIR.

Reply:
added simple verifier